packages/wallet/api/test/test-middleware.js

```diff
@@ -0,0 +1,76 @@
+// @ts-check
+import { test } from '@agoric/zoe/tools/prepare-test-env-ava.js';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { E, Far } from '@endo/far';
+import {
+  makeExportContext,
+  makeImportContext,
+  makeLoggingPresence,
+} from '../src/marshal-contexts.js';
+
+const capData1 = {
+  body: JSON.stringify([
+    [
+      'applyMethod',
+      { '@qclass': 'slot', iface: 'Alleged: purse.actions', index: 0 },
+      'deposit',
+      [{ '@qclass': 'slot', iface: 'Alleged: payment', index: 1 }],
+    ],
+    ['applyFunction', { '@qclass': 'slot', index: 0 }, [1, 'thing']],
+  ]),
+  slots: ['purse:1', 'payment:1'],
+};
+
+test('makeLoggingPresence logs calls on purse/payment actions', async t => {
+  const msgs = [];
+  const enqueue = m => msgs.push(m);
+  const purse = {
+    actions: await makeLoggingPresence('Alleged: purse.actions', enqueue),
+  };
+  const myPayment = Far('payment', {});
+  await E(purse.actions).deposit(myPayment); // promise resolves???
+  await E(purse.actions)(1, 'thing');
+  t.deepEqual(msgs, [
+    ['applyMethod', purse.actions, 'deposit', [myPayment]],
+    ['applyFunction', purse.actions, [1, 'thing']],
+  ]);
+
+  const ctx = makeExportContext();
```

Comment:
I'm confused about this test case, because `ctx` I thought would be inside `lib-wallet`, while the logging presence would be used outside `lib-wallet`

Reply:
Good point; I guess I'll think this over some more.

Reply:
I'd like to come back to this in a later PR.